plotly/tests/test_optional/test_figure_factory/test_figure_factory.py

```diff
@@ -54,41 +56,38 @@ def test_simple_distplot_prob_density(self):
         expected_dp_layout = {'barmode': 'overlay',
                               'hovermode': 'closest',
                               'legend': {'traceorder': 'reversed'},
-                              'xaxis1': {'anchor': 'y2', 'domain': [0.0, 1.0], 'zeroline': False},
-                              'yaxis1': {'anchor': 'free', 'domain': [0.35, 1], 'position': 0.0},
-                              'yaxis2': {'anchor': 'x1',
+                              'yaxis2': {'anchor': 'x',
                                          'domain': [0, 0.25],
                                          'dtick': 1,
                                          'showticklabels': False}}
-        self.assertEqual(dp['layout'], expected_dp_layout)
+        self.assert_dict_equal(dp.to_plotly_json()['layout'], expected_dp_layout)
```

Comment:
we might want to put `to_plotly_json` inside the `assert_fig_equal` method as well, that way we're not repeating this everywhere

Reply:
Good idea. @jmmease I believe you said you had plans to return an instance of the JSON when a graph_objs figure was instantiated correct? If this is the case, then I'd wait for that so then we can only remove the `.to_plotly_json`s throughout the tests

Reply:
@Kully The `Figure` class itself is going to stay a non-`dict` class, so we will need to use `to_plotly_json()` to compare it to dictionaries.  Moving it into `assert_fig_equal` makes sense to me.